### PR TITLE
feat: update cypress e2e tests to typescript

### PR DESCRIPTION
**Pull request by mrmodise**
Follow up to #476 based on https://github.com/strongloop/loopback4-example-shopping/pull/886#issuecomment-712296492

### CODE REVIEW

**Review by dougal83 — APPROVED** [review]
Nice.  All checks out! :+1: 

FYI: One test is a bit inconsistent, failed on first run.
**Profile: should update user profile**
`Timed out retrying: Expected to find content: 'LoopBack Doe' within the element: <div#user.nav-item> but never did.`

**Comment by mrmodise** [comment]
> Nice. All checks out! 👍
> 
> FYI: One test is a bit inconsistent, failed on first run.
> **Profile: should update user profile**
> `Timed out retrying: Expected to find content: 'LoopBack Doe' within the element: <div#user.nav-item> but never did.`

Thanks am going to add `.should('be.visible')` this should resolve it. The fun of using pop ups 😄

**Comment by dougal83** [comment]
> Thanks am going to add `.should('be.visible')` this should resolve it. The fun of using pop ups 😄

I think that would do it but how about a value oriented check?  Would make the test more self descriptive. What do you think?https://docs.cypress.io/guides/references/assertions.html#Value

**Comment by mrmodise** [comment]
> https://docs.cypress.io/guides/references/assertions.html#Value

Doesn't work. Returns the error immediately:

`expected <div#user.nav-item> to have value LoopBack Doe, but the value was ''`

Interesting that it times out:

`Timed out retrying: expected '<div#user.nav-item>' to have value 'LoopBack Doe', but the value was ''`

**Comment by dougal83** [comment]
> > https://docs.cypress.io/guides/references/assertions.html#Value
> 
> Doesn't work. Returns the error immediately:
> 
> `expected <div#user.nav-item> to have value LoopBack Doe, but the value was ''`
> 
> Interesting that it times out:
> 
> `Timed out retrying: expected '<div#user.nav-item>' to have value 'LoopBack Doe', but the value was ''`

I'm guessing that something like would work: `.then(() => cy.get('[data-cy=user]').should('contain', 'LoopBack Doe')),` Sorry, I didn't look in detail at markup before comment.

File: packages/shopping/src/__tests__/e2e/integration/profile.spec.ts
Comment (dougal83) [inline packages/shopping/src/__tests__/e2e/integration/profile.spec.ts:31]:
```suggestion
              .should('contain', 'Loopback Doe'),
```

File: packages/shopping/src/__tests__/e2e/integration/profile.spec.ts
Comment (dougal83) [inline packages/shopping/src/__tests__/e2e/integration/profile.spec.ts:32]:
```suggestion
              
```

File: packages/shopping/src/__tests__/e2e/integration/profile.spec.ts
Comment (mrmodise) [reply packages/shopping/src/__tests__/e2e/integration/profile.spec.ts]: Done

**Review by nabdelgadir — APPROVED** [review]
Changes LGTM 👍 The profile test is passing for me now as well.

**Comment by nabdelgadir** [comment]
The CI failure is unrelated to your changes. Can you push again so it restarts? Thanks!

**Comment by dhmlau** [comment]
Thanks @mrmodise. You're granted with write access to this repo. So feel free to land PRs in this repo as long as it has approvals. Thanks!

**Comment by mrmodise** [comment]
> Thanks @mrmodise. You're granted with write access to this repo. So feel free to land PRs in this repo as long as it has approvals. Thanks!

Thanks didn't want to seem am approving my own work 🤣

**Pull request merged**
